app/scripts/metamask-controller.js

```diff
@@ -2969,15 +2969,29 @@ export default class MetamaskController extends EventEmitter {
   /* eslint-disable accessor-pairs */
   /**
    * A method for recording whether the MetaMask user interface is open or not.
-   * @private
    * @param {boolean} open
    */
   set isClientOpen(open) {
+    if (!open) {
+      this.onPopUpClose();
```

Comment:
Just thought of something: if we are calling this method from here, then `onPopupClose` is not quite the right name. Because this can be called on close of other contexts as well, which is what we want.